### PR TITLE
Persistent login cookie

### PR DESCRIPTION
**Pull request by rrobgill**
**By submitting this pull request, I confirm the following:** `{please fill any appropriate checkboxes, e.g: [X]}`
- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [x] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#git-commit---signoff))

---

**What does this PR aim to accomplish?:**
Allow user to (optionally) use a cookie for automatic login.

`{A detailed description, screenshots (if necessary), as well as links to any relevant GitHub issues}`

At any normal login, it allows the user the option to set a cookie. (Default behaviour remains unchanged). Once the cookie is set, it is checked for before the login screen, and if present and valid, logs the user in automatically.

Clicking logout clears the cookie, so password will need to be entered again to login in future.

Expiry is set for 7 days and the password itself is not stored unhashed.

**How does this PR accomplish the above?:**
In loginpage.php, adding a checkbox and description of the option to the user.
In password.php, allowing the cookie to serve in place of the password on the login screen, clearing the cookie when the user selects logout. 

`{A detailed description (such as a changelog) and screenshots (if necessary) of the implemented fix}`

**What documentation changes (if any) are needed to support this PR?:**
Add information to the documentation for the login screen
`{A detailed list of any necessary changes}`

This is my commit message

Signed-off-by: rob gill <rrobgill@protonmail.com>

### CODE REVIEW

**Comment by DL6ER** [comment]
I'm unsure if, from a legal point of view, we also need an email address in the sign-off message. I'll try to find this out or someone from our team might now and comment here.

**Comment by rrobgill** [comment]
Cookie name has been changed to "persistentlogin" and user-facing note about setting cookies has been removed.

Mangled email address has been fixed.

**Comment by dschaper** [comment]
The DCO is part of the contributors, https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project but everything looks fine now. We can just ignore the bot comments, it's not a blocking check.

File: scripts/pi-hole/php/password.php [hunk 1]
```diff
@@ -24,9 +24,13 @@
     }
 
     // If the user wants to log out, we free all session variables currently registered
+    // and delete any persistent cookie.
     if(isset($_GET["logout"]))
     {
         session_unset();
+        setcookie('persistentlogin', '');
+	    header('Location: index.php');
```
Comment (Mcat12) [inline scripts/pi-hole/php/password.php]: Indentation is odd here.

File: scripts/pi-hole/php/password.php [hunk 2]
```diff
@@ -35,8 +39,23 @@
     // Test if password is set
     if(strlen($pwhash) > 0)
     {
+        // Check for and authorize from persistent cookie 
+        if (isset($_COOKIE["persistentlogin"]))
+        {
+            if ($pwhash = $_COOKIE["persistentlogin"])
+	    {	$auth = true;
+		// Refresh cookie with new expiry
+                setcookie('persistentlogin', $pwhash, time()+60*60*24*7);
```
Comment (Mcat12) [inline scripts/pi-hole/php/password.php]: Indentation is again odd.

File: scripts/pi-hole/php/password.php (hunk 1 above)
Comment (rrobgill) [reply scripts/pi-hole/php/password.php]: Fixed, thanks.

File: scripts/pi-hole/php/password.php (hunk 2 above)
Comment (rrobgill) [reply scripts/pi-hole/php/password.php]: Fixed, thanks.

File: scripts/pi-hole/php/password.php [hunk 3]
```diff
@@ -35,8 +39,23 @@
     // Test if password is set
     if(strlen($pwhash) > 0)
     {
+        // Check for and authorize from persistent cookie 
+        if (isset($_COOKIE["persistentlogin"]))
+        {
+            if ($pwhash = $_COOKIE["persistentlogin"])
+            { $auth = true;
+		// Refresh cookie with new expiry
+                setcookie('persistentlogin', $pwhash, time()+60*60*24*7);
```
Comment (Mcat12) [inline scripts/pi-hole/php/password.php]: There's still a mixture of spaces and tabs here, and there should not be anything besides the brace on line 46.

File: scripts/pi-hole/php/loginpage.php [hunk 4]
```diff
@@ -35,6 +35,9 @@
           </ul>
           </div>
           <div class="col-xs-4">
+            <div class="form-group">
+              <div class="checkbox pull-right"><label><input type="checkbox" id="logincookie" name="persistentlogin">Remember me for 7 days</label></div>
+              </div>
```
Comment (Mcat12) [inline scripts/pi-hole/php/loginpage.php]: Fix the indentation of line 40.

File: scripts/pi-hole/php/password.php (hunk 3 above)
Comment (rrobgill) [reply scripts/pi-hole/php/password.php]: done

File: scripts/pi-hole/php/loginpage.php (hunk 4 above)
Comment (rrobgill) [reply scripts/pi-hole/php/loginpage.php]: done

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/add-an-option-to-stay-logged-in-customise-what-boxes-are-on-dashboard-for-non-admins/2906/2

**Pull request merged**